.github/workflows/build_centos7.yml

```diff
@@ -26,16 +26,22 @@ jobs:
       - uses: actions/checkout@master
         with:
           fetch-depth: 0
-      - name: Get changed files
-        id: changed-files
-        uses: tj-actions/changed-files@v45
-        with:
-          files: |
-            docker/centos7-system-deps
+      - name: Dockerfile has changed
+        run: |
+          if ${{ github.event_name == 'pull_request' }}; then
+            changed_files=$(git diff --name-only -r HEAD^1 HEAD | xargs)
```

Comment:
`HEAD^1` is HEAD's leftmost parent of level 1, so probably not what you want here.

Also, `fetch-depth: 0` on `actions/checkout` might interfere with the diff command.